src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
It wouldn't be for developer use, but for performance. For instance, if you have 5 nodes and one is down, there is a 20% chance of selecting the down node for an operation, trying to execute the operation and losing time through that entire process before selecting a different node. Over the duration of time that the node is down that time adds up fast.